core/trino-main/src/main/java/io/trino/sql/planner/ConnectorExpressionTranslator.java

```diff
@@ -807,8 +806,7 @@ private ConnectorExpression constantFor(Expression node)
                     ImmutableMap.of());
 
             if (type == JONI_REGEXP) {
-                Slice pattern = ((JoniRegexp) value).pattern();
-                return new Constant(pattern, createVarcharType(countCodePoints(pattern)));
```

Comment:
this was here intetnionally. the JONI_REGEXP is considered engine's implementation detail and therefore we wanted connectors not to know about it